### PR TITLE
migrate LSR scripts

### PR DESCRIPTION
**Pull request by geforcelive**

### CODE REVIEW

**Comment by Daniel-Fan** [comment]
Hi @geforcelive 

May I ask that could we change the base branch to [`scripts-dev` branch](https://github.com/IBM/ibm-common-service-operator/tree/scripts-dev)?

File: cp3pt0-deployment/common/migrate_singleton.sh [hunk 1]
```diff
@@ -64,12 +65,52 @@ function main() {
...
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: 1Gi
+            storageClassName: "$LSR_STORAGE_CLASS"
+            volumeMode: Filesystem
+            volumeName: $LSR_PV_NAME
+EOF
+          ${OC} create -f TEMP_LSR_PVC_FILE
+          lsr_pvc_status=$("${OC}" get pvc license-service-reporter-pvc -n $LSR_NAMESPACE --no-headers | awk '{print $2}')
```
Comment (pgodowski) [inline cp3pt0-deployment/common/migrate_singleton.sh:105]: why we're checking the status but we're not using it anywhere?

File: cp3pt0-deployment/setup_singleton.sh [hunk 2]
```diff
@@ -232,6 +240,66 @@ function wait_for_license_instance() {
...
+
+  #create_reporter_instance
+  TEMP_LSR_FILE="_TEMP_LSR_FILE.yaml"
+
+  cat <<EOF >TEMP_LSR_FILE
+    apiVersion: operator.ibm.com/v1alpha1
+    kind: IBMLicenseServiceReporter
+    metadata:
+      name: ibm-lsr-instance
+      namespace: $LSR_NAMESPACE
```
Comment (pgodowski) [inline cp3pt0-deployment/setup_singleton.sh:283]:
```suggestion
      namespace: ${LSR_NAMESPACE}
```

File: cp3pt0-deployment/common/migrate_singleton.sh [hunk 3]
```diff
@@ -64,12 +65,52 @@ function main() {
...
+          LSR_STORAGE_CLASS=${OC} get pv -l license-service-reporter-pv=true -o=jsonpath='{.items[0].spec.storageClassName}'
+          # create PVC
+          TEMP_LSR_PVC_FILE="_TEMP_LSR_PVC_FILE.yaml"
+
+          cat <<EOF >TEMP_LSR_PVC_FILE
+          apiVersion: v1
+          kind: PersistentVolumeClaim
+          metadata:
+            name: license-service-reporter-pvc
+            namespace: $LSR_NAMESPACE
```
Comment (pgodowski) [inline cp3pt0-deployment/common/migrate_singleton.sh:93]:
```suggestion
            namespace: ${LSR_NAMESPACE}
```

File: cp3pt0-deployment/common/migrate_singleton.sh [hunk 4]
```diff
@@ -64,12 +65,52 @@ function main() {
...
+          spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: 1Gi
+            storageClassName: "$LSR_STORAGE_CLASS"
```
Comment (pgodowski) [inline cp3pt0-deployment/common/migrate_singleton.sh:100]:
```suggestion
            storageClassName: "${LSR_STORAGE_CLASS}"
```

File: cp3pt0-deployment/common/migrate_singleton.sh [hunk 5]
```diff
@@ -64,12 +65,52 @@ function main() {
...
+            volumeMode: Filesystem
+            volumeName: $LSR_PV_NAME
+EOF
+          ${OC} create -f TEMP_LSR_PVC_FILE
```
Comment (pgodowski) [inline cp3pt0-deployment/common/migrate_singleton.sh:104]: not consistent use of curly braces for env variable reference

**Review by pgodowski — CHANGES_REQUESTED** [review]

**Pull request closed without merging**